From 86cc3029d75f25913b38f3aac76c4453bd1038f1 Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Thu, 7 Jan 2021 14:55:14 +0800
Subject: [PATCH] drivers: rkflash: fixes compile error when !CONFIG_MTD

In file included from drivers/rkflash/sfc_nand_mtd.c:6:0:
./include/linux/mtd/cfi.h:76:2: warning: #warning No CONFIG_MTD_CFI_Ix selected. No NOR chip support can work. [-Wcpp]
error, forbidden warning:cfi.h:76
 #warning No CONFIG_MTD_CFI_Ix selected. No NOR chip support can work.
  ^~~~~~~
scripts/Makefile.build:333: recipe for target 'drivers/rkflash/sfc_nand_mtd.o' failed
make[2]: *** [drivers/rkflash/sfc_nand_mtd.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from drivers/rkflash/sfc_nor_mtd.c:6:0:
./include/linux/mtd/cfi.h:76:2: warning: #warning No CONFIG_MTD_CFI_Ix selected. No NOR chip support can work. [-Wcpp]
error, forbidden warning:cfi.h:76

Change-Id: I900d20adbd86c8293a9496ffba4bd722a46bfeae
Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
---
 drivers/rkflash/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/rkflash/Kconfig b/drivers/rkflash/Kconfig
index b84ad9eb6ce1..27600b3cdfa9 100644
--- a/drivers/rkflash/Kconfig
+++ b/drivers/rkflash/Kconfig
@@ -44,7 +44,7 @@ config RK_SFC_NAND
 config RK_SFC_NAND_MTD
 	bool "RK SFC NAND mtd Interface Support"
 	default n
-	depends on RK_SFC_NAND
+	depends on RK_SFC_NAND && MTD
 	help
 	  Enable mtd interface for SFC_NAND device.
 
@@ -65,7 +65,7 @@ config RK_SFC_NOR
 config RK_SFC_NOR_MTD
 	bool "RK SFC NOR mtd Interface Support"
 	default n
-	depends on RK_SFC_NOR
+	depends on RK_SFC_NOR && MTD
 	help
 	  Enable mtd interface for SFC_NOR device.
 
-- 
2.35.3

